sdk/storage/azure-storage-blob/azure/storage/blob/_shared/shared_access_signature.py

```diff
@@ -0,0 +1,368 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import sys
+from datetime import date
+
+from .constants import X_MS_VERSION
+from .utils import _sign_string, url_quote, _QueryStringConstants
+
+
+if sys.version_info < (3,):
+    def _str(value):
+        if isinstance(value, unicode):  # pylint: disable=undefined-variable
+            return value.encode('utf-8')
+
+        return str(value)
+else:
+    _str = str
+
+
+def _to_utc_datetime(value):
+    return value.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+
+class SharedAccessSignature(object):
+    '''
+    Provides a factory for creating account access
+    signature tokens with an account name and account key. Users can either
+    use the factory or can construct the appropriate service and use the
+    generate_*_shared_access_signature method directly.
+    '''
+
+    def __init__(self, account_name, account_key, x_ms_version=X_MS_VERSION):
+        '''
+        :param str account_name:
+            The storage account name used to generate the shared access signatures.
+        :param str account_key:
+            The access key to generate the shares access signatures.
+        :param str x_ms_version:
+            The service version used to generate the shared access signatures.
+        '''
+        self.account_name = account_name
+        self.account_key = account_key
+        self.x_ms_version = x_ms_version
+
+    def generate_account(self, resource_types, permission, expiry, start=None,
```

Comment:
It seems we should rename the method to something like generate_account_sas_token?

Reply:
Thanks! These are private/internal methods, copied from the track1 SDK. I will look at refactoring them/rename as needed for a subsequent preview!